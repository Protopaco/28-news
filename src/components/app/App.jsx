import React from 'react';
import NewsSearch from '../containers/NewsSearch/NewsSearch.jsx';

export default function App() {
  return (
    <>
      <NewsSearch />
    </>
  )
}
