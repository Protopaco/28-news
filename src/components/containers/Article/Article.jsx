import React from "react"
import PropTypes from 'prop-types';


const Article = ({ author, title, description }) => {

    return (
        <div>
            <hr />
            {author ?
                <p>{`author: ${author}`}</p>
                :
                <></>
            }
            <p>{`title: ${title}`}</p>
            <p>{`description: ${description}`}</p>
        </div>
    )

}

Article.propTypes = {
    author: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string
}

export default Article;
