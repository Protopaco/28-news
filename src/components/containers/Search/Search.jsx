import React, { useState } from "react"

const Search = ({ OnSearchSubmit }) => {
    const [searchInput, setSearchInput] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault();
        OnSearchSubmit(searchInput);
    }

    return (
        <form >
            <input
                data-testid='searchBar'
                onChange={(e) => { setSearchInput(e.target.value) }}
            />
            <button
                onClick={(e) => handleSubmit(e)}
                data-testid='submitButton'> Submit </button>
        </form >
    )
}


export default Search;
