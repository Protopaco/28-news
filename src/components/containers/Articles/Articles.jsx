import React from "react"
import Article from "../Article/Article.jsx"




const Articles = ({ articleList }) => {

    return (
        <div>
            {articleList ?
                articleList.map((article, index) => {
                    return <Article key={index} {...article} />
                })
                :
                <p>Loading!</p>
            }
        </div>
    )

}

export default Articles;