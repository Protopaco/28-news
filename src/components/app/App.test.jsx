import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import Articles from '../containers/Articles/Articles'
import Article from '../containers/Article/Article'
import Search from '../containers/Search/Search'

describe('App component', () => {
    afterEach(() => cleanup());

    it('renders Articles', () => {
        const { asFragment } = render(<Articles />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders Article', () => {
        const { asFragment } = render(<Article />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders Search', () => {
        const { asFragment } = render(<Search />);
        expect(asFragment()).toMatchSnapshot();
    });
});