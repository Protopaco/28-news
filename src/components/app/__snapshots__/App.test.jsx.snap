// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders Article 1`] = `
<DocumentFragment>
  <div>
    <hr />
    <p>
      title: undefined
    </p>
    <p>
      description: undefined
    </p>
  </div>
</DocumentFragment>
`;

exports[`App component renders Articles 1`] = `
<DocumentFragment>
  <div>
    <p>
      Loading!
    </p>
  </div>
</DocumentFragment>
`;

exports[`App component renders Search 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="searchBar"
    />
    <button
      data-testid="submitButton"
    >
       Submit 
    </button>
  </form>
</DocumentFragment>
`;
